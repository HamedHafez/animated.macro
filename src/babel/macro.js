const createMacroWithRuntime = require("./utils/createMacroWithRuntime");

module.exports = createMacroWithRuntime(
  ({ references, state, babel }) => {
    const transform = require("./transform");
    return transform(babel.types, references);
  },
  (...args) => {
    const runtime = require("../runtime");
    return runtime(...args);
  }
);
