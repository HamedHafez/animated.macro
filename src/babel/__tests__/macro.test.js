import { Animated } from "react-native";
import macro from "../macro";
const { Node: AnimatedNode } = Animated;
import pluginTester from "babel-plugin-tester";
import plugin from "babel-plugin-macros";

describe("macro", () => {
  test("falls back to runtime", () => {
    expect(macro`1`).toBeInstanceOf(AnimatedNode);
  });
});

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: { filename: __filename },
  tests: require("./expressions").map(
    src => `
      import { Animated } from "react-native";
      import animated from "../macro";
      
      ${src}
  `
  )
});
