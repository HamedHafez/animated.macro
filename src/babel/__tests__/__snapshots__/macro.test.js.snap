// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros 1. macros: 1. macros 1`] = `
"
import { Animated } from \\"react-native\\";
import animated from \\"../macro\\";

animated\`\${Math.PI} + 2\`

      ↓ ↓ ↓ ↓ ↓ ↓

var _Math$PI;

import { Animated } from \\"react-native\\";


Animated.add((_Math$PI = Math.PI, typeof _Math$PI === \\"number\\" ? new Animated.Value(_Math$PI) : _Math$PI), new Animated.Value(2));
"
`;

exports[`macros 2. macros: 2. macros 1`] = `
"
import { Animated } from \\"react-native\\";
import animated from \\"../macro\\";

const x = 1; animated\`\${x} + 2 + 3\`.interpolate()

      ↓ ↓ ↓ ↓ ↓ ↓

import { Animated } from \\"react-native\\";


const x = 1;Animated.add(Animated.add(new Animated.Value(x), new Animated.Value(2)), new Animated.Value(3)).interpolate();
"
`;

exports[`macros 3. macros: 3. macros 1`] = `
"
import { Animated } from \\"react-native\\";
import animated from \\"../macro\\";

animated\`1 + 1\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { Animated } from \\"react-native\\";


Animated.add(new Animated.Value(1), new Animated.Value(1));
"
`;

exports[`macros 4. macros: 4. macros 1`] = `
"
import { Animated } from \\"react-native\\";
import animated from \\"../macro\\";

animated\`3 % 2\`

      ↓ ↓ ↓ ↓ ↓ ↓

import { Animated } from \\"react-native\\";


Animated.modulo(new Animated.Value(3), 2);
"
`;

exports[`macros 5. macros: 5. macros 1`] = `
"
import { Animated } from \\"react-native\\";
import animated from \\"../macro\\";

animated\`\${foo} * 3\`

      ↓ ↓ ↓ ↓ ↓ ↓

var _foo;

import { Animated } from \\"react-native\\";


Animated.multiply((_foo = foo, typeof _foo === \\"number\\" ? new Animated.Value(_foo) : _foo), new Animated.Value(3));
"
`;

exports[`macros 6. macros: 6. macros 1`] = `
"
import { Animated } from \\"react-native\\";
import animated from \\"../macro\\";

animated\`(\${foo} + \${foo})\`

      ↓ ↓ ↓ ↓ ↓ ↓

var _foo, _foo2;

import { Animated } from \\"react-native\\";


Animated.add((_foo = foo, typeof _foo === \\"number\\" ? new Animated.Value(_foo) : _foo), (_foo2 = foo, typeof _foo2 === \\"number\\" ? new Animated.Value(_foo2) : _foo2));
"
`;
